src/StudioComponents/ScrollFrame/init.lua

```diff
@@ -1,467 +1,271 @@
--- Roact version by @sircfenner
--- Ported to Fusion by @YasuYoshida
+-- Written by @boatbomber
 
 local Plugin = script:FindFirstAncestorWhichIsA("Plugin")
 local Fusion = require(Plugin:FindFirstChild("Fusion", true))
 
 local StudioComponents = script.Parent
 local StudioComponentsUtil = StudioComponents:FindFirstChild("Util")
 
-local ScrollArrow = require(script.ScrollArrow)
-local ScrollBarHandle = require(script.ScrollBarHandle)
-
 local getState = require(StudioComponentsUtil.getState)
 local themeProvider = require(StudioComponentsUtil.themeProvider)
 local unwrap = require(StudioComponentsUtil.unwrap)
 local types = require(StudioComponentsUtil.types)
+local stripProps = require(StudioComponentsUtil.stripProps)
 
 local scrollConstants = require(script.Constants)
+local ScrollArrow = require(script.ScrollArrow)
 
-local Children = Fusion.Children
 local Computed = Fusion.Computed
-local OnChange = Fusion.OnChange
 local Hydrate = Fusion.Hydrate
-local OnEvent = Fusion.OnEvent
 local Value = Fusion.Value
 local New = Fusion.New
-local Spring = Fusion.Spring
-
-local BAR_SIZE = scrollConstants.ScrollBarSize
-local SCROLL_STEP = scrollConstants.ScrollStep
+local Children = Fusion.Children
+local OnChange = Fusion.OnChange
+local Ref = Fusion.Ref
+local Out = Fusion.Out
 
 local COMPONENT_ONLY_PROPERTIES = {
-	"Enabled",
-	"OnScrolled",
-	"ScrollingDirection",
-	"UIPadding",
-	"Layout",
-	"ZIndex",
+	"ScrollingEnabled",
+	"VerticalScrollBarPosition",
+	"VerticalScrollBarInset",
 	Children,
 }
 
-local function maxVector(vec, limit)
-	return Vector2.new(math.max(vec.x, limit.x), math.max(vec.y, limit.y))
-end
-
-local function clampVector(vec, min, max)
-	return Vector2.new(math.clamp(vec.x, min.x, max.x), math.clamp(vec.y, min.y, max.y))
-end
-
-local defaultLayout = {
-	SortOrder = Enum.SortOrder.LayoutOrder,
-}
-
-local baseProperties = {
-	BorderSizePixel = 1,
-	ScrollingDirection = Enum.ScrollingDirection.Y,
-}
-
 type ScrollFrameProperties = {
 	Enabled: (boolean | types.StateObject<boolean>)?,
-	OnScrolled: ((newCanvasPosition: Vector2) -> nil)?,
-	ScrollingDirection: (Enum.ScrollingDirection | types.StateObject<Enum.ScrollingDirection>)?,
-	UIPadding: UIPadding?,
-	Layout: UILayout?,
+	ScrollingEnabled: (boolean | types.StateObject<boolean>)?,
+	VerticalScrollBarPosition: (EnumItem | types.StateObject<EnumItem>)?,
+	VerticalScrollBarInset: (EnumItem | types.StateObject<EnumItem>)?,
```

Comment:
should use `Enum.ScrollBarInset` instead of being vague with `EnumItem`. The same goes for the EnumItem type above, except it'd be `Enum.VerticalScrollBarPosition`.